### PR TITLE
chore(MeshHTTPRoute): add query params match support

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? -- https://github.com/kumahq/kuma/issues/5470
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests -- none
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests -- none
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshhttproute/api/v1alpha1/meshhttproute.go [hunk 1]
```diff
@@ -64,6 +67,20 @@ type PathMatch struct {
...
+type QueryParamsMatchType string
+
+const (
+	ExactQueryMatch             QueryParamsMatchType = "Exact"
+	RegularExpressionQueryMatch QueryParamsMatchType = "RegularExpression"
+)
+
+type QueryParamsMatch struct {
+	Type  QueryParamsMatchType `json:"type"`
+	Name  string               `json:"name"`
```
Comment (slonka) [inline pkg/plugins/policies/meshhttproute/api/v1alpha1/meshhttproute.go:80]: should have min len  = 1

File: pkg/plugins/policies/meshhttproute/xds/configurer.go [hunk 2]
```diff
@@ -125,6 +122,46 @@ func (c RoutesConfigurer) routeMethodMatch(envoyMatch *envoy_route.RouteMatch, m
...
+				},
+			}
+		case api.RegularExpressionQueryMatch:
+			matcher = envoy_type_matcher.StringMatcher{
+				MatchPattern: &envoy_type_matcher.StringMatcher_SafeRegex{
+					SafeRegex: regexMatcher(match.Value),
+				},
+			}
+		default:
+			panic("impossible")
```
Comment (slonka) [inline pkg/plugins/policies/meshhttproute/xds/configurer.go:151]: nitpick: do we always panic on these things? Let's say someone makes a bug in the future in validation this and it crash loops in prod for someone - not sure it's the right approach :(

**Review by slonka — CHANGES_REQUESTED** [review]
The "min length" must be changed, the other stuff let's discuss in comments.

File: pkg/plugins/policies/meshhttproute/xds/configurer.go (hunk 2 above)
Comment (michaelbeaumont) [reply pkg/plugins/policies/meshhttproute/xds/configurer.go:151]: Yeah it's a good question. If someone adds a new enum variant and it gets released, no matter what we do here, that new variant/functionality won't work. And in order to stop the error/panic, the user need only remove the new variant.

So either way, we must keep the implications of adding a new variant in mind during review. 

At least with a panic, the code can stay clean (no useless `error` return value) and the fact that the functionality is broken is immediately obvious to users and thus to us.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Review by slonka — APPROVED** [review]

**Pull request merged**